### PR TITLE
feat: add smoke test in cli example

### PR DESCRIPTION
**Pull request by felixfaisal**
To run this go `cli` and run the following 
```
cargo run --examples request_vc_smoke_test
```

### CODE REVIEW

File: tee-worker/cli/examples/request_vc_smoke_test.rs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+			assertion,
+			Some(key),
+			Default::default(),
+		)
+		.sign(&KeyPair::Sr25519(Box::new(alice)), nonce, &mrenclave, &shard)
+		.into_trusted_operation(trusted_cli.direct);
+
+		let start = Instant::now();
+
+		// TODO: P-177, print actual VC content to stdout
```
Comment (Kailai-Wang) [inline tee-worker/cli/examples/request_vc_smoke_test.rs:143]: This issue is done :)

File: tee-worker/cli/examples/request_vc_smoke_test.rs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	}
+}
+
+#[tokio::main]
+async fn main() -> Result<(), String> {
+	println!("Starting smoke test!");
+	let cli = generate_dummy_cli();
+	let trusted_cli = generate_dummy_trusted_cli();
+	let cli = Arc::new(cli);
+	let trusted_cli = Arc::new(trusted_cli);
```
Comment (Kailai-Wang) [inline tee-worker/cli/examples/request_vc_smoke_test.rs:184]: We can combine them I assume?

File: tee-worker/cli/examples/request_vc_smoke_test.rs [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+			self.identity,
+			self.substrate_identity,
+			Some(pair.public().into()),
+			self.prime_pair,
+			None,
+			Some(pair),
+		)
+	}
+
+	// TODO: Construct request_vc function
```
Comment (kziemianek) [inline tee-worker/cli/examples/request_vc_smoke_test.rs:115]: Can this todo be resolved?

File: tee-worker/cli/examples/request_vc_smoke_test.rs [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+}
+
+#[tokio::main]
+async fn main() -> Result<(), String> {
+	println!("Starting smoke test!");
+	let cli = generate_dummy_cli();
+	let trusted_cli = generate_dummy_trusted_cli();
+	let cli = Arc::new(cli);
+	let trusted_cli = Arc::new(trusted_cli);
+	for x in 0..10 {
```
Comment (Kailai-Wang) [inline tee-worker/cli/examples/request_vc_smoke_test.rs:185]: Why do we need 10 iterations? If it's for smoke test, will just one attempt suffice?

If we do need multiple rounds, maybe we should add it as command line arg

File: tee-worker/cli/examples/request_vc_smoke_test.rs (hunk 1 above)
Comment (kziemianek) [inline tee-worker/cli/examples/request_vc_smoke_test.rs:143]: This issue is already resolved.

**Review by Kailai-Wang — COMMENTED** [review]
I haven't fully understood it - is it supposed to work for prod worker too? What's different from using:
```
./bin/litentry-cli link-dientity ...
./bin/litentry-cli request-vc ...
```

File: tee-worker/cli/examples/request_vc_smoke_test.rs [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+	pub fn request_vc(&self, cli: &Cli, trusted_cli: &TrustedCli) -> Result<(), String> {
+		let alice = self.prime_pair.clone();
+		let id = self.identity.clone();
+
+		let (mrenclave, shard) = get_identifiers(trusted_cli, cli);
+		let nonce = get_layer_two_nonce!(alice, cli, trusted_cli);
+		let assertion = Assertion::A1;
+
+		let mut key = RequestAesKey::default();
+		hex::decode_to_slice(
```
Comment (kziemianek) [inline tee-worker/cli/examples/request_vc_smoke_test.rs:125]: reuse `COMMON_AES` ?

File: tee-worker/cli/examples/request_vc_smoke_test.rs [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+			nonce: 0,
+			prime_pair,
+			substrate_pair,
+			evm_pair,
+			user_shielding_key_set: false,
+			link_identity: false,
+		}
+	}
+
+	pub fn create_substrate_identity_with_junction(self, junction: u32) -> User {
```
Comment (kziemianek) [inline tee-worker/cli/examples/request_vc_smoke_test.rs:83]: What's using this function ?

**Review by kziemianek — COMMENTED** [review]
I tried to run it locally but it failed.

command `cargo run --example request_vc_smoke_test`

output:

```
Starting smoke test!
Encountered an error: None
Enable a logger to see more information.
thread 'main' panicked at 'Unable to retrieve MRENCLAVE from endpoint: MspcReceiver(RecvError)', cli/src/trusted_command_utils.rs:90:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Is the command I used to run it correct?

**Comment by felixfaisal** [comment]
I will raise this PR after rebasing with `dev` to accomodate new and breaking changes.

**Pull request closed without merging**